### ISSUE TITLE
App Crashes on Scanning QR Code in Friends Activity 

### ISSUE THREAD

**Issue by vedang14** [named-data-mobile/ndn-photo-app#99]
On Scanning QR code on Friends Activity app crashes immediately, I propose to add a popup for not valid referenced QR Code, and create a list( history specific) for corresponding scanned QR in a separate list view from which each one of them could be accessed and deleted if needed. 
Moreover I also wish to add the flashlight and setting functions on the QR Code activity layout through use of Camera API 2 that provides in turn more flexibility and code stabilization.

**Comment by vedang14** [comment]
@shiv07tiwari @jrandallclark  sir, please do let me know your thoughts about it and the added functionalities as I have already successfully implemented one of them.

**Comment by dev-ritik** [comment]
Basically, there's no check in the pattern of information decoded from any QRcode scanner in the app as of now.

**Comment by jrandallclark** [comment]
Yes, keeping the app from crashing when a non-valid QR code is scanned is something that need to be implemented.

But I don't understand what you mean be creating a history specific list for corresponding scanned QRs. 

And by flashlight, do you mean the camera's flash?

**Comment by vedang14** [comment]
Ok sir, so I'll add on this functionality to prevent crash on non valid QR, and raise PR for the same.

Sir by history specific I mean that whenever we successfully scan one's QR code we should be able to save there reference(name or any other attribute) in a list accordingly in a specific STACK manner, from where we could access it again and delete it if required.

And by flashlight i mean ( Camera's Flash ).

**Comment by vibhusharma101** [comment]
@jrandallclark  sir can I work on this issue?

**Comment by jrandallclark** [comment]
Yes, you may.
